### DISCUSSION TITLE
Welcome to idris-jvm Discussions!

### DISCUSSION THREAD

**Discussion by mmhelloworld** [mmhelloworld/idris-jvm#113] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nightscape** [comment]
Hi all, opening the round here 😃 
I'm Martin, a Scala programmer.
I'm waiting for a chance to use Idris in a real-world project, and idris-jvm might play an essential part in that as it opens the door to a plethora of JVM libraries (preferably functional ones).

**Comment by mmhelloworld** [comment]
Hi @nightscape ,

Welcome! I would be excited to see Idris being used more as well and of course on the JVM too. I am currently working on bootstrapping Idris 2 on the JVM as well as adding a JVM bytecode backend [here](https://github.com/mmhelloworld/Idris2-boot). Good news is that the compiler can already run on the JVM and compile few simple programs to JVM bytecode with major items like TCO already in place. 

Once the bootsrapping is done along with a JVM backend, next step would be to compile current [Idris 2](https://github.com/idris-lang/Idris2) with this and adding a JVM bytecode backend for current Idris 2. As we can see, there is still lot of work to do and I am going to keep working on this. Hopefully soon, we can catch up with current Idris 2 repository for the JVM and then the focus would be to improve the libraries. 

Everyone is welcome to share new ideas, suggestions or pull requests (to [Idris2 Boot](https://github.com/mmhelloworld/Idris2-boot) for now) to make Idris 2 possible on the JVM. This repository (idris-jvm) for Idris 1 is now in mostly read-only mode except for fixes to compile Idris 2 on the JVM. Once Idris 2 bootstrap and Idris 2 repositories are ready for the JVM, this repository code would be replaced with Idris 2 code for the JVM.

**Reply by dysinger** [reply]
yay
